### DISCUSSION TITLE
Netbox API Tokens Permissions

### DISCUSSION THREAD

**Discussion by gfhoffmann** [netbox-community/netbox#11821] [Help Wanted!] [answered]
Hello there! I'm trying to create custom permissions to the action of creating/deleting/updating/viewing API Tokens, but it seems that even a new user with no permissions at all can have all access of token creation.

I tried disabling the token permissions, added constraints to the token permissions, but none works, what I'm missing here?

Thanks!

**Comment by kkthxbye-code** [comment, marked as answer]
If you mean creating tokens for your own user in the UI, then that is not something you can turn off. The API token has the same access as the user has in the UI.

**Reply by gfhoffmann** [reply]
Ah, that explain the issue! I was trying to disable the access in the UI. Thanks for the response!

**Reply by kkthxbye-code** [reply]
Out of curiosity why do you want to disable access for users to create API tokens? They don't provide any additional access, so I'm having a hard time coming up with a good reason.

**Reply by Leo-scs** [reply]
We wanted to restrict it so that only one user can generate API tokens to centralize management and know where the tokens are being used.

**Reply by esoubihe** [reply]
Even if they don't provide additional access, leaking an API token is incredibly easy. We can control user auth with SAML but can't do that for APi keys.

**Comment by mtinberg** [comment]
While this may not solve your problem the way you were planning, there is an API endpoint which can see user tokens /api/users/tokens/, so you can audit who has tokens and report on changes or DELETE tokens if necessary.  The URL for token management /user/api-tokens/ could also be blocked at the http proxy level if need be, or restricted based on group if you are doing auth in the webserver, right?


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
